xwiki-platform-core/xwiki-platform-office/xwiki-platform-office-importer/src/main/java/org/xwiki/officeimporter/internal/filter/ImageFilter.java

```diff
@@ -179,8 +179,19 @@ private String getFileName(Attr source) throws MimeTypeException
             int separator = value.lastIndexOf('/');
             fileName = separator < 0 ? value : value.substring(separator + 1);
             try {
+
+                // the + character is not escaped in HtmlCleaner as it is an authorized character
+                // but when decoding it for URL it would be replaced by a space
+                // so we encode it manually here.
+                fileName = fileName.replaceAll("\\+", "%2B");
                 // We have to decode the image file name in case it contains URL special characters.
                 fileName = URLDecoder.decode(fileName, UTF_8);
+
+                // we also need to replace the "&amp;" by "&" since HtmlCleaner replace them automatically to &amp;
+                fileName = fileName.replaceAll("&amp;", "&");
+
+                // '@' must also be escaped in order to resolve properly the path in XWiki
+                fileName = fileName.replaceAll("@", "\\\\@");
```

Comment:
Could have been good to explain *why* by listing both:
* List of special characters that need to be escaped in XML attributes
* List of special characters that need to be URL-encoded in URL

Then explain that HC URL-encodes all characters it finds in HTML attributes (that's a known limitation). Woudl be good to open an issue on HC issue tracker and reference it so that we can follow the progress and remove stuff no longer needed in the future.

Thus we need to unencode characters that are special chars for XML attributes and that are also special chars for URLs.

WDYT?